### PR TITLE
Attempt to merge develop into branch

### PR DESCRIPTION
**Pull request by GhettoKiwi**
Signed-off-by: Benjamin Mønster <bem@bankdata.dk>
Made by @sam94dion

### CODE REVIEW

File: ALLTESTS [hunk 1]
```diff
@@ -1,331 +1,8 @@
-      *From file: C:\Udvikler\Workspace\cobol-check-main\src\test\cobol\MOCKTEST\BeforeAfterTest.cut
```
Comment (Rune-Christensen) [inline ALLTESTS:1]: Kan vi tilføje denne til gitignore?
Indeholder den ikke bare data fra sidste test kørsel?

File: src/main/java/org/openmainframeproject/cobolcheck/features/testSuiteParser/Keywords.java [hunk 2]
```diff
@@ -153,12 +153,12 @@ public class Keywords {
                 new Keyword(Constants.TO_EQUAL_KEYWORD,
                         Arrays.asList(),
                         new HashMap<String, List<String>>() {{
-                            put(Constants.EXPECT_KEYWORD, Arrays.asList(
-                                    Constants.NUMERIC_KEYWORD,
-                                    Constants.ALPHANUMERIC_LITERAL_KEYWORD,
-                                    Constants.NUMERIC_LITERAL_KEYWORD,
-                                    Constants.BOOLEAN_VALUE,
-                                    Constants.COBOL_TOKEN)
+                                put(Constants.EXPECT_KEYWORD, Arrays.asList(
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/testSuiteParser/Keywords.java:156]: Could these be moved back to original position, so they are without additional indentation?

Comment (Rune-Christensen) [reply src/main/java/org/openmainframeproject/cobolcheck/features/testSuiteParser/Keywords.java:156]: This is for all the changes in this file, regarding additional indentation.

File: CC##99.CBL [hunk 3]
```diff
@@ -1,7 +1,7 @@
        IDENTIFICATION DIVISION.                                                 
-       PROGRAM-ID.  MOCKTEST.
```
Comment (Rune-Christensen) [inline CC##99.CBL:2]: Could this also be added to .gitignore?

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 4]
```diff
@@ -23,86 +24,132 @@ public class InterpreterController {
...
             throw new PossibleInternalLogicErrorException(
-                    Messages.get("ERR001", "cobolSourceIn", "Generator.runSuite()"));
-        }
-        reader = new CobolReader(sourceReader);
-        lineRepository = new LineRepository();
-        numericFields = new NumericFields();
-        tokenExtractor = new StringTokenizerExtractor();
-        currentDataStructure = new TreeMap<>();
+                Messages.get("ERR001", "cobolSourceIn", "Generator.runSuite()"));
+            }
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:35]: These indentations should be restored to their original position.

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 5]
```diff
@@ -116,109 +163,117 @@ public boolean isCurrentLineEndingSectionOrParagraph() throws IOException {
...
+        else if (Interpreter.endsInPeriod(reader.getCurrentLine())) {
+            reader.putNextLine("           .");
+            return false;
+        }
         return false;
 
     }
 
-    /**Interprets the next line from the source file. Based on the line, the following values
+    /**
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:187]: More comments that should be restored to original position.

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 6]
```diff
@@ -116,109 +163,117 @@ public boolean isCurrentLineEndingSectionOrParagraph() throws IOException {
...
     }
 
-     /**Updates dependencies of the interpreter.
-      * The following values can be updated, based on the given line:
+    /**
+     * Updates dependencies of the interpreter.
+     * The following values can be updated, based on the given line:
      * - Current statement (only if multiline statement was read)
-     * - The state of the reader (flags that shows what part of the program we are reading)
+     * - The state of the reader (flags that shows what part of the program we are
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:250]: Another comment that should be moved back.

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 7]
```diff
@@ -238,42 +293,46 @@ private void updateReplaceStatement(CobolLine line) {
...
             possibleMockArgs = Interpreter.getUsingArgs(statement);
         }
 
         if (possibleMockIdentifier != null && periodShouldBeOnThisLine
                 && !Interpreter.endsInPeriod(line)
-                && Interpreter.containsOnlyPeriod(nextLine)){
-            //We might generate code after the current line, thus if the period is on the next line,
-            //we append it to this line. This prevents us generating code in the wrong place.
+                && Interpreter.containsOnlyPeriod(nextLine)) {
+            // We might generate code after the current line, thus if the period is on the
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:332]: A couple of comments that should be restored to original.

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 8]
```diff
@@ -286,23 +345,29 @@ private void updatePossibleStub(CobolLine line) throws IOException {
...
             }
 
         }
     }
-    /**If the current line is a numeric field, it will be added to the list of numeric fields.
-     * We need to save field names of numeric data items in the DATA DIVISION in case a test case
-     * references them. There's no way to distinguish numeric fields while reading the PROCEDURE
+
+    /**
+     * If the current line is a numeric field, it will be added to the list of
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:365]: More comment fixes

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 9]
```diff
@@ -335,61 +401,81 @@ private void updateNumericFields(CobolLine line) {
...
+     * multiple lines. If the line fits any of below it will be saved in the
+     * repository:
      * - It is a line read under FILE CONTROL
-     * - It has a SELECT token. File identifier and corresponding FILE STATUS will be saved
+     * - It has a SELECT token. File identifier and corresponding FILE STATUS will
+     * be saved
      * - It contains a copy token
-     * - It is a file section statement (file section statements from referenced copybooks are also included)
+     * - It is a file section statement (file section statements from referenced
+     * copybooks are also included)
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:428]: More comment fixes

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 10]
```diff
@@ -335,61 +401,81 @@ private void updateNumericFields(CobolLine line) {
...
+            for (int i = 0; i < extractedCopyBook.size(); i++) {
+                CobolLine cobolLine = new CobolLine(extractedCopyBook.get(i), tokenExtractor);
+                updateNumericFields(cobolLine);
+            }
+        }
     }
 
     /**
-     * If the given line contains a SELECT token, the file identifier will be added, waiting for
+     * If the given line contains a SELECT token, the file identifier will be added,
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:465]: Comments

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java [hunk 11]
```diff
@@ -398,31 +484,33 @@ private void updateLineRepoBySelectToken(CobolLine line){
     }
 
     /**
-     * If the given line contains a file status token, it will be added to a map to a
+     * If the given line contains a file status token, it will be added to a map to
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/InterpreterController.java:487]: Comments

File: src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java [hunk 12]
```diff
@@ -14,34 +14,34 @@ public class Interpreter {
 
     // Source tokens from Procedure Division that begin batch I/O statements
     private static final List<String> batchFileIOVerbs = Arrays.asList(
-            "OPEN", "CLOSE", "READ", "WRITE", "REWRITE", "DELETE", "START"
-    );
+            "OPEN", "CLOSE", "READ", "WRITE", "REWRITE", "DELETE", "START");
 
-    // Used for handling source lines from copybooks that may not have the standard 80-byte length
+    // Used for handling source lines from copybooks that may not have the standard
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java:19]: Comment

File: src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java [hunk 13]
```diff
@@ -146,18 +146,18 @@ public static String setFlagsForCurrentLine(CobolLine line, CobolLine nextLine,
      * (b) - previous line contains just a period
      * (c) - first token on this line is a Cobol verb
      *
-     * @param currentLine - current source line being processed
+     * @param currentLine        - current source line being processed
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java:149]: Remove added indentation

File: src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java [hunk 14]
```diff
@@ -181,10 +182,12 @@ public static boolean lineEndsParagraphOrSection(CobolLine currentLine, CobolLin
      * This "shouldn't happen." Famous last words.
      *
      * @param line
-     * @return true if the source line is too short to be a meaningful line of code in Cobol.
+     * @return true if the source line is too short to be a meaningful line of code
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java:185]: comment

File: src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java [hunk 15]
```diff
@@ -301,32 +314,32 @@ private static boolean isBatchFileIOStatement(List<String> tokens, String ioVerb
...
      * source line against specific attributes that makes up a paragraph header.
      *
-     * @param line - The line to check
+     * @param line     - The line to check
      * @param nextLine - The line after the line param
-     * @param state - current state of flags
-     * @return true if the source line have all the attributes of a paragraph header.
+     * @param state    - current state of flags
+     * @return true if the source line have all the attributes of a paragraph
+     *         header.
```
Comment (Rune-Christensen) [inline src/main/java/org/openmainframeproject/cobolcheck/services/cobolLogic/Interpreter.java:340]: Comment fixes

**Review by dakaa16 — APPROVED** [review]
I will approve even though there are a lot of formatting that I do not agree with

**Review by ThomasNellemann — APPROVED** [review]
ok with me

**Review by dakaa16 — APPROVED** [review]

**Pull request merged**